app/views/comments/_threads.html.erb

```diff
@@ -1,25 +1,33 @@
 <%# locals: (thread:, story: nil, ribbon: nil) -%>
 <% previous_depth = nil %>
 <% top_comment_depth = nil %>
+<%# open list of threads %>
+<ol class="comments">
 <% thread.each do |comment| %>
 
-    <% if !previous_depth || comment.depth > previous_depth %>
-      <%# brittle - see li.comments_subtree > .comment:nth-last-child(2) in css %>
-      <span class="prior_comment_has_children"></span>
+    <% if previous_depth %>
+      <% if comment.depth > previous_depth %>
```

Comment:
Whew, this has gotten complicated, it took me a bit of reading to work out how you'd rearranged the branches and your comments helped a lot.

I've experimented with laying out comments using grid so they'd all be sibling `<li>` on a single `<ol>`. It'll need quite a bit of work to finish, but it would mean we don't have this big hairy code for performantly mapping the linear list of comments into a nested tree structure.

I've been really pleasantly surprised that anyone would tackle all this really hairy coupling on our hot path, so thank you, and if (after I merge this PR) you're interested in burning it all down, I'd be happy to set you up to try it.